src/OperatingSystemApp/controlpanel.js

```diff
@@ -1,10 +1,88 @@
-/* exported OSControlPanel */
+/* exported OSControlPanel, OSModel */
 
-const {GObject, Gtk} = imports.gi;
+const {Gio, GObject, Gtk} = imports.gi;
+const {Codeview} = imports.codeview;
+const {Lockscreen} = imports.lockscreen;
+
+GObject.type_ensure(Codeview.$gtype);
+GObject.type_ensure(Lockscreen.$gtype);
+
+var _propFlags = GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT;
+
+var OSModel = GObject.registerClass({
+    Properties: {
+        wobblyEffect: GObject.ParamSpec.boolean(
+            'wobblyEffect', 'Wobbly Effect', '',
+            _propFlags, false),
+        wobblyObjectMovementRange: GObject.ParamSpec.double(
+            'wobblyObjectMovementRange', 'Wobbly Object Movement Range', '',
+            _propFlags, 10.0, 500.0, 100.0),
+        wobblySlowdownFactor: GObject.ParamSpec.double(
+            'wobblySlowdownFactor', 'Wobbly Slowdown Factor', '',
+            _propFlags, 1.0, 5.0, 1.0),
+        wobblySpringFriction: GObject.ParamSpec.double(
+            'wobblySpringFriction', 'Wobbly Spring Friction', '',
+            _propFlags, 2.0, 10.0, 3.0),
+        wobblySpringK: GObject.ParamSpec.double(
+            'wobblySpringK', 'Wobbly Spring K', '',
+            _propFlags, 2.0, 10.0, 8.0),
+    },
+}, class OSModel extends GObject.Object {
+    _init(props = {}) {
+        super._init(props);
+
+        this._settings = new Gio.Settings({
+            schemaId: 'org.gnome.shell',
+            path: '/org/gnome/shell/',
+        });
+
+        this._keys = this._settings.settings_schema.list_keys();
+
+        this.bindSetting('wobbly-effect', this, 'wobblyEffect');
+        this.bindSetting('wobbly-object-movement-range', this, 'wobblyObjectMovementRange');
+        this.bindSetting('wobbly-slowdown-factor', this, 'wobblySlowdownFactor');
+        this.bindSetting('wobbly-spring-friction', this, 'wobblySpringFriction');
+        this.bindSetting('wobbly-spring-k', this, 'wobblySpringK');
+    }
+
+    bindSetting(setting, object, property) {
```

Comment:
This can be private (start with an underscore)

Reply:
yeah I forgot to make it private (I was using outside the class before)